### PR TITLE
History & like video

### PR DESCRIPTION
**Pull request by animeshgarai09**
Done with History and video like, dislike feature.
Small file updates are there, you can ignore those. just check provider files and new files.

Thank you

### CODE REVIEW

File: src/components/side-nav/side-nav.js [hunk 1]
```diff
@@ -14,23 +14,26 @@ import {
...
 
 const SideNav = () => {
+    const handleActiveNav = (navigationData) => {
+        return navigationData.isActive ? styles.active : ""
+    }
     return (
         <aside className={styles.container}>
             <List style="none">
                 <ListItem className={styles.nav_link}>
-                    <NavLink className={(navigationData) => navigationData.isActive ? styles.active : null} to="/explore"><IconButton icon={<MdOutlineExplore />} className={styles.icon} /></NavLink>
```
Comment (govindappaarun) [inline src/components/side-nav/side-nav.js:21]: missing prettier / code formatter, line length crossing 80

File: src/pages/history/history.js [hunk 2]
```diff
@@ -1,18 +1,31 @@
...
+                    history.length !== 0
+                        ? [...history].reverse().map((item, i) => {
+                            return <VideoCard
+                                orientation="horizontal"
+                                videoItem={item}
+                                key={i}
+                                onRemove={removeFromHistory}
+                            />
+                        })
+                        : [...Array(4)].map((_, i) => {
```
Comment (govindappaarun) [inline src/pages/history/history.js:25]: when this will timeout?, right now it looks loading forever

File: src/components/header/header.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ import { ReactComponent as Logo } from "@assets/svg/felix.svg"
 const Header = () => {
     const [drop, setDrop] = useState(false)
     const { UserState, handleLogout } = useAuth();
+    console.log("header")
     return (
         <header className={styles.container}>
             {/* <img src="favicon/favicon-32x32.png" alt="" /> */}
```
Comment (MayankKumar10) [inline src/components/header/header.js:16]: remove this comment

File: src/store/providers/history-provider.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                })
+                navigate('/signin', { state: { from: location }, replace: true })
+            })
+        }
+    }
+
+    const removeFromHistory = (id) => {
+        axios.delete(`/api/user/history/${id}`, {
+            headers: {
+                authorization: encodedToken,
```
Comment (govindappaarun) [inline src/store/providers/history-provider.js:47]: suggestion:  to use interceptors to add token in one place

File: src/store/providers/history-provider.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+import axios from "axios"
+import { useNavigate, useLocation } from 'react-router-dom'
+import { useToast } from 'react-felix-ui'
+
+const HistoryContext = createContext()
+
+const initState = {
+    history: [],
+    count: 0,
+    // playTime: 0,
```
Comment (MayankKumar10) [inline src/store/providers/history-provider.js:11]: remove this comment too

File: src/store/providers/like-provider.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            headers: {
+                authorization: encodedToken,
+            },
+        }).then((response) => {
+            setLikesState({ ...LikesState, likes: response.data.likes });
+        }).catch(err => {
+            console.log(err);
+            toast({
+                status: "error",
+                message: "Sign in to your account first",
```
Comment (govindappaarun) [inline src/store/providers/like-provider.js:57]: This toast can be put conditionally, api failure doesn't always mean invalid token

**Review by MayankKumar10 — APPROVED** [review]
Look into these suggestions else everything looks good to me you can merge it.

**Review by govindappaarun — APPROVED** [review]
I have few suggestions to address,
Otherwise  rest of things looks good to merge

File: src/components/side-nav/side-nav.js (hunk 1 above)
Comment (animeshgarai09) [reply src/components/side-nav/side-nav.js:21]: yeah not sure why it does that

File: src/pages/history/history.js (hunk 2 above)
Comment (animeshgarai09) [reply src/pages/history/history.js:25]: The state changes for map once the data is updated, there is a bug for infinite loading. will be fixing that soon. thanks for noticing.

File: src/store/providers/history-provider.js (hunk 4 above)
Comment (animeshgarai09) [reply src/store/providers/history-provider.js:47]: Not sure what you ar suggesting

File: src/store/providers/like-provider.js (hunk 6 above)
Comment (animeshgarai09) [reply src/store/providers/like-provider.js:57]: okay sure. will make the changes

File: src/store/providers/history-provider.js (hunk 5 above)
Comment (VLeads) [inline src/store/providers/history-provider.js:11]: Unnecessary comments can be removed !

**Review by VLeads — APPROVED** [review]
I have gone through your entire code. Not much to add
But your code is clean and easily understandable.
It's good wherever required you have chunked piece of code into components.
It's good that you are using module.css.
You have nice job in CSS work for the UI.
Keep doing the great work !!

**Pull request merged**